src/features/scheduled-messages.ts

```diff
@@ -1,85 +1,152 @@
-import * as discord from "discord.js";
-import cron from "node-cron";
+import type * as discord from "discord.js";
+import { CHANNELS } from "../constants";
+import { logger } from "./log";
+import { scheduleTask } from "../helpers/schedule";
 
-export const MESSAGE_SCHEDULE: MessageConfig[] = [
+const HOURLY = 60 * 60 * 1000;
+// By keeping these off 24 hr, we can make sure they show up at all timezones. If
+// it were 24 hours, for instance, it would consistently show up in the middle of
+// the night for some timezones.
+const DAILY = 20 * HOURLY;
+const FREQUENCY = {
+  often: 9 * HOURLY,
+  daily: DAILY,
+  moreThanWeekly: 2 * DAILY,
+  weekly: 6 * DAILY,
+};
+
+type MessageConfig = {
+  postTo: {
+    guildId?: discord.Snowflake;
+    interval: number;
+    channelId: discord.Snowflake;
+  }[];
+  message: discord.MessageOptions;
+};
+const MESSAGE_SCHEDULE: MessageConfig[] = [
   /*  Example:
   {
-    cronExpression: "0,30 * * * *",  // https://crontab.guru/#0,30_*_*_*_*
-    guilds: [
+    // Find Discord channel IDs: https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-
+    postTo: [
       {
-        // getting these IDs: https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-
-        id: "102860784329052160",  // Reactiflux's server ID
-        channelIds: ["103696749012467712"]  // #help-react
+        id: "102860784329052160",  // Reactiflux's server ID, optional
+        interval: FREQUENCY.weekly, // Frequency the bot should post by
+        channelIds: [ CHANNELS.helpReact ]  // Add channel IDs to constants first!
       }
     ],
     message: {
-      embed: {
-        title: "Example Message",
-        description: "This message is posted every 0th and 30th minute of the hour"
-      }
+      content: "A message to post, any type of message discord.js understands"
     }
   }
   */
-];
+  {
+    postTo: [{ interval: FREQUENCY.daily, channelId: CHANNELS.jobBoard }],
+    message: {
+      content: `Messages must start with [FORHIRE]/[HIRING]. Check the channel description for a full list of tags and rules!
 
-export type MessageConfig = {
-  cronExpression: string;
-  guilds: { id: discord.Snowflake; channelIds: discord.Snowflake[] }[];
-  message: discord.MessageOptions;
-};
+* Job postings may only be posted every 7 days.
+* Posts should be reasonably descriptive.
+* Jobs are paid — unpaid, equity-only, or similar are not allowed.
+* We don't allow "small gigs".
+
+Moderators may remove posts at any time, with or without warning. Repeat violators of these rules will be removed from the server permanently.
+`,
+    },
+  },
+  {
+    postTo: [{ interval: FREQUENCY.often, channelId: CHANNELS.helpJs }],
```

Comment:
While I see the benefit for the jobBoard and helpReact channels I don't really see what benefit the helpJs and helpReact brings.

Maybe we should improve the new joiner experience in general? e.g. have a channel with a message explaining all of this visible right after joining and having other channels hidden for like 2 minutes, giving everyone time to read?

Reply:
I'd quite like to improve the new joiner experience in general, adding some form of onboarding would be awesome — right now we can't control what channel people enter, so I thought repeatedly surfacing info that that's tucked away in channel descriptions was a low-effort way to improve on the current state

Reply:
We can always update the welcome screen channels (up to 5 supported).